cylc/flow/clean.py

```diff
@@ -326,13 +326,16 @@ def _clean_using_glob(
         LOG.info(f"No files matching '{pattern}' in {run_dir}")
         return
     # First clean any matching symlink dirs
-    for path in abs_symlink_dirs:
-        if path in matches:
-            remove_dir_and_target(path)
-            if path == run_dir:
+    for symlink_dir in abs_symlink_dirs:
+        # Note: must clean e.g. share/cycle/ before share/ if the former
+        # is a symlink even if only the latter was specified.
+        if any(is_relative_to(symlink_dir, path) for path in matches):
```

Comment:
@oliver-sanders RE: the traceback you were getting, can you try this in your testing?
```suggestion
        if (
            any(is_relative_to(symlink_dir, path) for path in matches)
            and symlink_dir.is_symlink()
        ):
```

Reply:
Seems to do the trick!